code/modules/reagents/reactions/other.dm

```diff
@@ -245,6 +245,19 @@
 	s.start()
 	holder.clear_reagents()
 
+/datum/chemical_reaction/razorburn
+	name = "Razorburn Gas"
+	required_reagents = list(/datum/reagent/foaming_agent = 1, /datum/reagent/toxin/nanites2 = 1)
+
+/datum/chemical_reaction/razorburn/on_reaction(datum/reagents/holder, created_volume)
+	var/location = get_turf(holder.my_atom)
+	for(var/mob/M in viewers(WORLD_VIEW_NUM, location))
+		to_chat(M, "<span class='danger'>The solution spews out a dense, ground-hugging gas! Get away!</span>")
```

Comment:
use visible_message

Reply:
Can do.

Reply:
visible_message apparently is an undefined proc. Swapping back to working version, any ideas?

Reply:
use location.visible_message(
since it needs a atom source

Reply:
seems easy enough, will do shortly.

Reply:
It done brokeded. ~but at least I fixed another horrible issue caused by my own stupidity!~

Reply:
thx 4 help, fixed